packages/shared-preferences/android/src/main/java/io/flutter/plugins/shared_preferences/SharedPreferencesPlugin.java

```diff
@@ -49,42 +102,47 @@ private SharedPreferencesPlugin(Activity activity) {
   @Override
   public void onMethodCall(MethodCall call, MethodChannel.Result result) {
     String key = call.argument("key");
-    switch (call.method) {
-      case "setBool":
-        editor.putBoolean(key, (boolean) call.argument("value")).apply();
-        result.success(null);
-        break;
-      case "setDouble":
-        editor.putFloat(key, (float) call.argument("value")).apply();
-        result.success(null);
-        break;
-      case "setInt":
-        editor.putInt(key, (int) call.argument("value")).apply();
-        result.success(null);
-        break;
-      case "setString":
-        editor.putString(key, (String) call.argument("value")).apply();
-        result.success(null);
-        break;
-      case "setStringSet":
-        editor.putStringSet(key, new HashSet<>((List<String>) call.argument("value"))).apply();
-        result.success(null);
-        break;
-      case "commit":
-        result.success(editor.commit());
-        break;
-      case "getAll":
-        result.success(getAllPrefs());
-        break;
-      case "clear":
-        for (String keyToDelete : getAllPrefs().keySet()) {
-          editor.remove(keyToDelete);
-        }
-        result.success(editor.commit());
-        break;
-      default:
-        result.notImplemented();
-        break;
+    try {
+      switch (call.method) {
+        case "setBool":
+          editor.putBoolean(key, (boolean) call.argument("value")).apply();
+          result.success(null);
+          break;
+        case "setDouble":
+          editor.putFloat(key, (float) call.argument("value")).apply();
+          result.success(null);
+          break;
+        case "setInt":
+          editor.putInt(key, (int) call.argument("value")).apply();
+          result.success(null);
+          break;
+        case "setString":
+          editor.putString(key, (String) call.argument("value")).apply();
+          result.success(null);
+          break;
+        case "setStringList":
```

Comment:
Note: technically, we could support JSON-typed maps now as well using this (I assume `NSUserDefaults` supports storing `NSDictionary`...).  In any case, not worth adding in this PR.